### DISCUSSION TITLE
QuantUMS

### DISCUSSION THREAD

**Discussion by NicoDrou** [vdemichev/DiaNN#764] [Q&A]
Dear Vadim,

I am very interested by the new QuantUMS quantification you implemented in the DIA-NN 1.8.2 beta22. 
I have finished to process a batch of 200 samples with it and i would have a few questions:
1) using the same search parameters and the PG report with no additional filtering, dia-NN 1.8.2 beta 22 identifies ~ 540 protein groups whereas the 1.8.1 version found only ~440 protein groups. Is it normal? have you made any modification in the search algorithm that is not explained in the preprint?
2) It is not clear to me what quantification mode you recommend. From what I understand of the preprint, the high accuracy should be more suited as the quantification is more accurate. And an extra filter step based on the new quality metrics would discard the low-quality precursors and proteins. Is it correct?
3) I am new to R, but I understand we need to use the .parquet file to filter the data because we need the new PG.MaxLFQ.Quality to filter the proteins and the Quantity.Quality to filter the precursors.
5) The column Quantity.Quality already existed in the 1.8.1 version. Is it still the same? Would you recommend filtering data from the 1.8.1 version using this parameter?
6) in my data set, i find precursors with no MS1.Area but with a a Quantity.Quality >0.75, sometimes >0.9. How is it possible? I would have expected precursors with undetectable MS1 peaks to have a very low score. Would you then recommend also filtering using the MS1.Area?
3) concerning the protein quantification. In the main report of the 1.8.2 beta22, the maxLFQ column still exists. I guess this is an artefact, but the new quantUMS algorithm is used. is it correct? 
4) will you release a quantUMS R script to convert precursors to protein intensity the same way you did for the maxLFQ?

Many thanks in advance for your time and all your amazing work :)

Regards,
Nicolas

**Comment by LeeSCantrell** [comment]
To add a datapoint on 1.8.2 b22, I find search somewhat slower on 1.8.2 with Thermo Data (2-15%). Precursor IDs I see trend slightly down in 1.8.2 vs 1.8.1. Within rounding error for most files... 

I'm also unclear on how to return to legacy high precision robust LC mode since ```--peak-center``` is not recognized as a CLI option. 
Can we get clarity on how to enable non-composite quant methods seen in 1.8.1?

**Reply by vdemichev** [reply]
Hi Nicolas,

Thank you for your interest in QuantUMS :) 

1. Yes, newer version is better.
2. High-accuracy is probably better in most cases, but everyone loves low CVs, that's why high-precision is default.
3. Yes, it's very easy to load:
library(arrow)
df <- read_parquet("file name here")
4. There was a very simplistic attempt at quantity quality in previous DIA-NN versions, yes. But nothing even close to QuantUMS. Yes, you can filter the report in 1.8.1 using it too. Basically, low values mean that something is wrong with quantity, high values don't guarantee that it is good though. In QuantUMS, high values mean that LC-MS-related error in the quantity is very likely negligible.
5. Main quantification is done via MS2, hence MS1 signal not required. But QuantUMS does take advantage of MS1 when it is good. Some of IDs without MS1 are genuine IDs which just show poor MS1 signal, some are peptidoforms (e.g. peptide can have some modification but DIA-NN thinks it's unmodified).
6. In DIA-NN 'MaxLFQ' refers to protein quant via pariwise comparison of samples, this is also what is used in QuantUMS, except in QuantUMS it's done in a way that is uncertainty-aware, and hence brings benefits also at the protein quant level. But maybe you are right, we should rename to PG.QuantUMS, will think about it :)
7. First it will be a standalone C++ tool (open source) - soon, R or Python might follow but not specific timeline.

Also, we have since improved QuantUMS, and an updated version will be released soon.

Best,
Vadim

**Reply by vdemichev** [reply]
Hi Lee,

1.8.2 should actually be faster for dia-PASEF, one of its major improvements, not sure about Thermo.
Legacy (direct) is Robust LC (high-precision) equivalent of previous versions.

Best,
Vadim

**Reply by NicoDrou** [reply]
Dear Vadim,
thank you for your answer.

Initially, I was very skeptical about trusting DIA identification with no MS1, but now I understand your point with the peptidoforms with PTM assigned as an unmodified peptide. Thank you for pointing this out :)

I would still have a few questions:
1) So if I understand correctly, the MS1 level isn't used for quantification, but only as a quality marker. 
So if we want to work from the main report and apply custom filters, we can use the normalized intensity of the precursors to calculate the maxLFQ while waiting for the standalone C++ tool for QuantUMS. Is it correct? 

2a) Regarding the new metrics (Quantity.Quality and  PG.MaxLFQ.Quality), what filter would you recommend? If I understood correctly the preprint, you kept the top 25% (or 50%) of the precursors. Was it for both Quantity.Quality and  PG.MaxLFQ.Quality, only one of them?  
2b) Would a recommend a fixed filtering threshold?
2c) what filters are now applied in the output matrices concerning the Quantity.Quality and  PG.MaxLFQ.Quality?

3a) Just to be sure, in version 1.8.2 beta 22, for the composite high accuracy/high precision, ALL MaxLFQ associated values should be understood as QuantUMS associated value, for all matrices report, main report and parquet report?
3b) Is it also true for the legacy mode? 

4) I am probably asking too much, but by any chance do you have any estimation of the release date of the standalone C++ and the 1.8.2 versions :) 

Many thanks again for your help and the amazing work of you and your team!

Regards,
Nicolas

**Reply by vdemichev** [reply]
Hi Nicolas,

1.
- In QuantUMS MS1 is used for quant to generate Precursor.Quantity and Precursor.Normalised. In previous versions - no. In all versions separate Ms1.Area column is reported.
- QuantUMS is built-in DIA-NN, no need to wait for standalone version :)

2a. Really depends on the application. We see that for getting better DE numbers a mild filter on average PG.MaxLFQ.Quality helps.
2b. Basic filter like quality > 0.8 might also make sense.
2c. None.
3a. Yes.
3b. Legacy is regular MaxLFQ.
4. In a week or two.

Best,
Vadim

**Reply by NicoDrou** [reply]
1) 
- My apologies, but I am now confused. I understood from your previous answer ("_Main quantification is done via MS2, hence MS1 signal not required._ ") only the fragment were used for the Precursor.Quantity (Main quantification is done via MS2, hence MS1 signal not required. ). Then I guess just as for the fragment, the MS1 of a precursor is used for the Precursor,Quantity only if it is detected through all the samples. Correct?

2a) ok, so PG.MaxLFQ.Quality should be used as a filter rather than Quantity.Quality. What do you mean by DE numbers? 

I just noticed something and i think i understood something. Quantity.Quality is a precursor based metric whereas the PG.MaxLFQ.Quality is the protein group-based metric calculated using the Quantity.Quality of each precursor used in QuantUMS protein quantification. Is it correct?
Therefore, if I understand we would have 2 filtering approaches:
a) we directly go for the protein level in the parquet output and filter on the PG.MaxLFQ.Quality (and the regular lib.Q;value, lib.PG.Q.value, etc.) and do our data normalization process based on the PG.MaxLFQ (soon renamed PG.QuantUMS :D )
b) we normalize at the precursor level and filter using the Quality.Quantity (and the regular lib.Q;value, lib.PG.Q.value, etc.). Then we would need the standalone C++ QuantUMS tool to quantify the proteins using QuantUMS. Am I right? And i guess MaxLFQ would also work here, correct? Would you also use the quantile 0.5 to filter the precursor level with the Quantity.Quality?

one or two weeks !!!! that is awesome !!!!!!!!!!! 
Thank you so much for all your effort and your help ! :)

**Reply by vdemichev** [reply]
1. the MS1 of a precursor is used for the Precursor,Quantity only if it is detected through all the samples - no, there's a more complex way of doing it in QuantUMS.
2. differentially expressed protein numbers. Correct.
a) Yes.
b) Yes, this will be useful for precursor-level batch-correction.

**Comment by NicoDrou** [comment]
Dear Vadim,
Is there any update about the release date of the standalone C++ QuantUMS tool for the QuantUMS protein quantification from manually filtered precursors? :) 

Many thanks in advance.
Regards,
Nicolas

**Reply by vdemichev** [reply]
Hi Nicolas,

Will be released once the journal publication is out. Cannot really put an approximate date on it yet.

Best,
Vadim

**Reply by NicoDrou** [reply]
Unfortunate but I understand :)
Thank you ! I am looking forward the final publication and the stand-alone tool :)
